src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -241,7 +241,7 @@ private CSharpAttributeData GetAttribute(BoundAttribute boundAttribute, BindingD
             else
             {
                 constructorArguments = GetRewrittenAttributeConstructorArguments(out constructorArgumentsSourceIndices, attributeConstructor,
-                    constructorArgsArray, boundAttribute.ConstructorArgumentNamesOpt, (AttributeSyntax)boundAttribute.Syntax, diagnostics, ref hasErrors);
+                    constructorArgsArray, boundAttribute.ConstructorArgumentNamesOpt, (AttributeSyntax)boundAttribute.Syntax, boundAttribute.ConstructorArgumentsToParamsOpt, diagnostics, ref hasErrors);
```

Comment:
>ConstructorArgumentNamesOpt [](http://example.com/codeflow?start=57&length=27)

It feels like we should be able to completely get rid of this field in favor of using ```ConstructorArgumentsToParamsOpt```. However, it is probably better to follow up on that separately.